bin/add_frontend_server.rb

```diff
@@ -0,0 +1,93 @@
+require 'aws-sdk'
+require_relative '../deployment'
+
+MAX_WAIT_TIME = 600 #Wait no longer than 10 minutes for instance creation. Typically this takes a couple minutes
+
+ec2client = Aws::EC2::Client.new
+
+instances = ec2client.describe_instances
+
+#Determine distribution of availability zones, pick the one that has the least capacity among frontend instances
+frontend_instances = []
+
+instances.reservations.each do |reservation|
+  frontends_for_reservation = reservation.instances.select{|instance| instance.tags.detect{|tag| tag.key == 'Name' && tag.value.include?('frontend')}}
+  frontend_instances << frontends_for_reservation unless frontends_for_reservation.size == 0
+end
```

Comment:
You could use [`map`](http://ruby-doc.org/core-2.0.0/Enumerable.html#method-i-map) and [`flatten`](http://ruby-doc.org/core-2.0.0/Array.html#method-i-flatten) to simplify this part:
```
frontend_instances = instances.reservations.map do |reservation|
  reservation.instances.select{|instance| instance.tags.detect{|tag| tag.key == 'Name' && tag.value.include?('frontend')}}
end.flatten
```